CoreScriptsRoot/Modules/Settings/Pages/ReportAbuseMenu.lua

```diff
@@ -278,6 +278,34 @@ do
 	PageInstance.Displayed.Event:connect(function()
 		PageInstance:UpdatePlayerDropDown()
 	end)
+
+	function PageInstance:ReportPlayer(player)
+		if player then
+			local setReportPlayerConnection = nil
+			setReportPlayerConnection = PageInstance.Displayed.Event:connect(function()
+				-- When we change the SelectionIndex of GameOrPlayerMode it waits until the tween is done
+				-- before it fires the IndexChanged signal. The WhichPlayerMode dropdown listens to this signal
+				-- and resets when it is fired. Therefore we need to listen to this signal and set the player we want
+				-- to report the frame after the dropdown is reset
+				local indexChangedConnection = nil
+				indexChangedConnection = PageInstance.GameOrPlayerMode.IndexChanged:connect(function()
+					if indexChangedConnection then
+						indexChangedConnection:disconnect()
+						indexChangedConnection = nil
+					end
+					wait() -- We need to wait a frame to set the value of WhichPlayerMode as it is being updated by another script listening to the IndexChanged signal
```

Comment:
Now that we have moved this event callback to the ReportAbuseMenu.lua script can we remove the `wait()`? We are updating the `WhichPlayerMode` value in this same script. Perhaps this events can be folded together.